server/src/domain/asset/asset.service.ts

```diff
@@ -420,11 +427,54 @@ export class AssetService {
       await this.assetRepository.updateAll([options.stackParentId], { stackParentId: null });
     }
 
+    if (creationDate !== undefined || latitude !== undefined || longitude !== undefined) {
+      this.validateGpsCoordinates(latitude, longitude);
+      for (const id of ids) {
+        await this.updateExifData(id, { CreationDate: creationDate, GPSLatitude: latitude, GPSLongitude: longitude });
+      }
+    }
+
     await this.jobRepository.queue({ name: JobName.SEARCH_INDEX_ASSET, data: { ids } });
     await this.assetRepository.updateAll(ids, options);
     this.communicationRepository.send(CommunicationEvent.ASSET_UPDATE, authUser.id, ids);
   }
 
+  validateGpsCoordinates(latitude: number | undefined, longitude: number | undefined): void {
```

Comment:
Please remove all validation logic from the service, it should be handled in the dto itself. You can use `ValidateIf()` to conditionally run validation

Reply:
Ok it's done!